File: scripts/.local/bin/show_time.py
#!/usr/bin/env python

import sys
secs = int(sys.argv[1])
mins = secs // 60
hours = mins // 60
days = hours // 24

show_days = days > 0
show_seconds = not show_days

if show_days:
    print(f'{days}d', end=' ')
print(f'{hours % 24}h {mins % 60}m', end='')
if show_seconds:
    print(f' {secs % 60}s')
else:
    print()

File: scripts/.local/bin/show_time.cpp
#include <string>
#include <iostream>

int main(int argc, char* argv[]) {
	if (argc < 2) { return 1; }
	auto const seconds = std::stoul(argv[1]);
	auto const minutes = seconds / 60;
	auto const hours = minutes / 60;
	auto const days = hours / 24;
	if (days > 0) {
		std::cout << days << "d ";
	}
	std::cout << hours % 24 << "h " << minutes % 60 << 'm';
	if (days == 0) {
		std::cout << ' ' << seconds % 60 << 's';
	}
	std::cout << std::endl;
	return 0;
}
